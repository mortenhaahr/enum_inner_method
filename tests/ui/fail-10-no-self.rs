
use enum_inner_method::enum_inner_method;

pub struct Circle;
impl Circle {
    pub fn nothing() {}
}

pub struct Rectangle;
impl Rectangle {
    pub fn nothing() {}
}

#[enum_inner_method(fn nothing())]
pub enum ShapeEnum {
    Circle(Circle),
    Rectangle(Rectangle),
}

fn main() {
    let shapes = [ShapeEnum::Circle(Circle), ShapeEnum::Rectangle(Rectangle)];
}
