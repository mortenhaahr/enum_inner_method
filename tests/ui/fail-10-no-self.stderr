error: custom attribute panicked
  --> tests/ui/fail-10-no-self.rs:14:1
   |
14 | #[enum_inner_method(fn nothing())]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: In current version the first argument must be self

error[E0433]: failed to resolve: use of undeclared type `ShapeEnum`
  --> tests/ui/fail-10-no-self.rs:21:19
   |
21 |     let shapes = [ShapeEnum::Circle(Circle), ShapeEnum::Rectangle(Rectangle)];
   |                   ^^^^^^^^^ use of undeclared type `ShapeEnum`

error[E0433]: failed to resolve: use of undeclared type `ShapeEnum`
  --> tests/ui/fail-10-no-self.rs:21:46
   |
21 |     let shapes = [ShapeEnum::Circle(Circle), ShapeEnum::Rectangle(Rectangle)];
   |                                              ^^^^^^^^^ use of undeclared type `ShapeEnum`
